Diagnostic/DistroSpecific.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python
+#
+# Azure Linux extension
+# Distribution-specific actions
+#
+# Linux Azure Diagnostic Extension (Current version is specified in manifest.xml)
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+# MIT License
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the ""Software""), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import exceptions
+import time
+import subprocess
+import re
+from Utils.WAAgentUtil import waagent
+
+omi_universal_install_cmd = 'bash scx-1.6.2-337.universal.x64.sh --upgrade'
+
+
+class CommonActions:
+    def __init__(self, logger):
+        self.logger = logger
+        pass
+
+    def log_run_get_output(self, cmd):
+        """
+        Execute a command in a subshell
+        :param str cmd: The command to be executed
+        :return (int, str): A tuple of (subshell exit code, contents of stdout)
+        """
+        self.logger("RunCmd " + cmd)
+        error, msg = waagent.RunGetOutput(cmd, chk_err=True)
+        self.logger("Return " + str(error) + ":" + msg)
+        return int(error), str(msg)
+
+    def log_run_ignore_output(self, cmd):
+        """
+        Execute a command in a subshell
+        :param str cmd: The command to be executed
+        :return int: The subshell exit code
+        """
+        self.logger("RunCmd " + cmd)
+        error, msg = waagent.RunGetOutput(cmd, chk_err=True)
```

Comment:
Refactored so it just uses log_run_get_output() and discards the output. Added the new parameter.